docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,132 @@
+* **UC04** As a cluster administrator, I want to deploy APIMan and its components reusing existing infrastructure components where possible (e.g. ElasticSearch), so I can minimize infrastructure components.
+* **UC05** As a cluster administrator, I want to reuse the aggregated logging CA when generating certs for the APIMan components, so I can reuse the ElasticSearch cluster.
+* **UC06** As an API consumer, I want to configure my services to consume other services
+
+## Specification
+
+API services will be scoped to OpenShift projects. Projects will have one-to-one relationship to an APIMan namespace (formally organization). APIMan will utilize namespace
+to manage policy regarding the service.
+
+### Deployment Scenario - APIMan Gateway Fronted by a Router
+
+The primary deployment scenario is to deploy APIMan components as shown in **Figure 1**.  APIMan will be deployed as a cluster level service for managing API service end points. APIMan will be deployed to reuse the existing ElasticSearch cluster for storage.  Communication between APIMan and ElasticSearch will make use of the CA certificate that was used for aggregated logging.  Modifications to the aggregated logging deployer will be made to save the CA as a secret.  This will allow the APIMan deployer to reuse the certificate to create client certificates.
+
+
+**Figure 1 - APIMan Deployment Diagram**
+
+![APIMan Deployment Diagram](apiman_deployment_diagram.png "Figure 1")
+
+The APIMan gateway is deployed in conjunction with a supported OpenShift router.  A typical request to a managed endpoint is initially routed to the gateway and then further routed by the gateway to the targeted service.  This flow can be seen in **Figure 2** where the APIMan gateway relies upon the service proxy to find pods.
+
+**Figure 2 - Managed Service Request Flow with APIMan Gateway and Router**
+
+![APIMan request flow with router](apiman_request_wi_router.png "Figure 2")
+
+
+### Service Annotations
+Services intended to be managed and exposed as API Endpoints will be annotated<sup>[5](#r5)</sup>.  The annotations are repeated below for convenience:
+
+```
+  apiVersion: "v1"
+  kind: "Service"
+  metadata: 
+    annotations: 
+      api.service.kubernetes.io/protocol: REST
+      api.service.kubernetes.io/scheme: http
+      api.service.kubernetes.io/path: cxfcdi
+      api.service.kubernetes.io/description-path: cxfcdi/swagger.json
+      api.service.kubernetes.io/description-language: SwaggerJSON
+```
+Additionally, services that should be managed by APIMan will be further annotated identifying who is managing the service.  The proposed annotation is:
+```
+  openshift.io/service-managed-by: apiman
+```
+
+Service providers will need to explicitly publish a service using the APIMan user interface.  Future iterations may include functionality to automatically publish a service when these annotations are applied.
```

Comment:
What purpose do these annotations serve if they do not result in services being published? What is going to look at them?